pkg/api/deployment.go

```diff
@@ -34,21 +35,51 @@ func CreateContainers(app Application) []v12.Container {
 	for key, value := range app.Environment {
 		envVars = append(envVars,v12.EnvVar{ Name: key, Value: value })
 	}
+	
+	var volumeMounts []v12.VolumeMount
+	for _, volume := range app.Volumes {
+		for path := range volume {
+			volumeMounts = append(volumeMounts, v12.VolumeMount{
+				Name: CreatePVCName(app, path),
+				MountPath: path,
+			})
+		}
+	}
 
 	containers := []v12.Container{{
 		Name: app.Name,
-		Image: app.Image + ":" + app.Version,
+		Image: fmt.Sprintf("%s:%s", app.Image, app.Version),
 		Env: envVars,
-		VolumeMounts: nil,
+		VolumeMounts: volumeMounts,
 	}}
 	
 	return containers
 }
 
+func CreateVolumes(app Application) []v12.Volume {
+	volumes := make([]v12.Volume, 0)
```

Comment:
You could probably replace `0` with `len(app.Volumes)` and then uncomment `_` below and use the `i` index directly, then you wouldn't require the `append`.